### PR TITLE
[NONPR-2134] add diagnostic logging and integration tests

### PR DESCRIPTION
**Pull request by PaulHodgson**
```
sbt test
...
[info] ScalaTest
[info] Run completed in 11 seconds, 230 milliseconds.
[info] Total number of tests run: 55
[info] Suites: completed 13, aborted 0
[info] Tests: succeeded 55, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Passed: Total 55, Failed 0, Errors 0, Passed 55
[success] Total time: 24 s, completed 05-Jul-2021 11:44:49
```

```
set it:test 
...
[info] ScalaTest
[info] Run completed in 20 seconds, 358 milliseconds.
[info] Total number of tests run: 21
[info] Suites: completed 2, aborted 0
[info] Tests: succeeded 21, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Passed: Total 21, Failed 0, Errors 0, Passed 21
[success] Total time: 23 s, completed 05-Jul-2021 11:46:58
```

acceptance-tests
```
[info] ScalaTest
[info] Run completed in 3 minutes, 17 seconds.
[info] Total number of tests run: 0
[info] Suites: completed 0, aborted 0
[info] Tests: succeeded 0, failed 0, canceled 0, ignored 0, pending 0
[info] No tests were executed.
[info] Passed: Total 24, Failed 0, Errors 0, Passed 24
[success] Total time: 215 s (03:35), completed 05-Jul-2021 11:52:31
```

### CODE REVIEW

File: app/connectors/NrsRetrievalConnectorImpl.scala [hunk 1]
```diff
@@ -69,20 +68,22 @@ class NrsRetrievalConnectorImpl @Inject()(val environment: Environment,
   }
 
   override def statusSubmissionBundle(vaultName: String, archiveId: String)(implicit hc: HeaderCarrier): Future[HttpResponse] = {
```
Comment (PaulHodgson) [inline app/connectors/NrsRetrievalConnectorImpl.scala:70]: add logging

File: app/controllers/SearchController.scala [hunk 2]
```diff
@@ -145,10 +142,9 @@ class SearchController @Inject()(@Named("retrieval-actor") retrievalActor: Actor
   def doAjaxRetrieve(vaultName: String, archiveId: String): Action[AnyContent] = Action.async { implicit request =>
     logger.info(s"Request retrieval for $vaultName, $archiveId")
     authWithStride("Download", { user =>
-      ask(retrievalActor, SubmitMessage(vaultName, archiveId, hc, user)).mapTo[Future[ActorMessage]].flatMap(identity).map {
-        case _ =>
```
Comment (PaulHodgson) [inline app/controllers/SearchController.scala:149]: more logging

File: build.sbt [hunk 3]
```diff
@@ -56,6 +58,8 @@ lazy val root = (project in file("."))
     scalacOptions += "-P:silencer:pathFilters=target/.*",
     publishingSettings,
     scoverageSettings)
+  .settings(defaultSettings(): _*)
```
Comment (PaulHodgson) [inline build.sbt:61]: fix `it:test` config

File: it/uk/gov/hmrc/nrsretrievalfrontend/IntegrationSpec.scala [hunk 4]
```diff
@@ -0,0 +1,47 @@
+/*
```
Comment (PaulHodgson) [inline it/uk/gov/hmrc/nrsretrievalfrontend/IntegrationSpec.scala:1]: add the Integration Specs back

**Review by greg-gajda — APPROVED** [review]
looks good

**Pull request merged**